File: object/csharp/src/IPebbleDataSource.cs
using System.Collections.Generic;

namespace PebbleFields.PebbleObject
{
    /// <summary>
    /// </summary>
    public interface IPebbleDataSource
    {

			IPebbleDataSource getNewInstance();

			IPebbleDataSource getNewInstance(string str);

			IPebbleDataSource getDataSource(string key);

			IPebbleDataSource getCopyInstance();

			void removeSpaces();

			string getValue(string key);

			void setValue(string path, string val);

			void setMarkup(string path, string val);

			string getTagName();
			
			string getRecordSetAttribute(string path, string str);

			void setRecordSetAttribute(string path, string val, string attrName);

			void removeRecordSetAttribute(string path, string attrName);

			void setRecordSet(string path, IPebbleDataSource obj, bool copyProperties);

			IPebbleDataSource remove(string path);

			List<IPebbleDataSource> getRecords(string path);

			int getIndex(string path, string name);

			void add2Array(string path, IPebbleDataSource xml, string name);

			void insertBefore(IPebbleDataSource newNode, string targetPath);

			IPebbleDataSource getByIndex(string path, int index);

			void loadFile(string path);

			string toString();

    }
}

File: object/java/src/org/pebblefields/pebbleobject/IPebbleDataSource.java
package org.pebblefields.pebbleobject;

import java.util.ArrayList;

public interface IPebbleDataSource {

	IPebbleDataSource getNewInstance();

	IPebbleDataSource getNewInstance(String str);

	IPebbleDataSource getDataSource(String key);

	IPebbleDataSource getCopyInstance();

	void removeSpaces();

	String getValue(String key);

	void setValue(String path, String val);
	
	void setMarkup(String path, String val);
	
	String getTagName();
	
	String getRecordSetAttribute(String path, String string);

	void setRecordSetAttribute(String path, String val, String attrName);
	
	void removeRecordSetAttribute(String path, String attrName);
	
	void setRecordSet(String path, IPebbleDataSource obj, boolean copyProperties);

	IPebbleDataSource remove(String path);

	ArrayList<IPebbleDataSource> getRecords(String path);
	
	int getIndex(String path, String name);

	void add2Array(String path, IPebbleDataSource xml, String name);
	
	void insertBefore(IPebbleDataSource newNode, String targetPath);

	IPebbleDataSource getByIndex(String path, int index);

	void loadFile(String path);

}
